x-pack/plugins/lens/public/indexpattern_service/loader.ts

```diff
@@ -104,19 +104,36 @@ export function convertDataViewIntoLensIndexPattern(
     fields: newFields,
     getFieldByName: getFieldByNameFactory(newFields),
     hasRestrictions: !!typeMeta?.aggs,
-    spec: dataView.isPersisted() ? undefined : dataView.toSpec(false),
+    spec: dataView.toSpec(false),
+    isPersisted: dataView.isPersisted(),
   };
 }
 
 export async function loadIndexPatternRefs(
   dataViews: MinimalDataViewsContract,
-  adHocDataViews?: Record<string, DataViewSpec>
+  adHocDataViews?: Record<string, DataViewSpec>,
+  contextDataViewSpec?: DataViewSpec
 ): Promise<IndexPatternRef[]> {
   const indexPatterns = await dataViews.getIdsWithTitle();
+  const missedIndexPatterns = Object.values(adHocDataViews || {});
+
+  // add data view from context
+  if (contextDataViewSpec) {
+    const existedDataView = indexPatterns.find(
+      (indexPattern) => indexPattern.id === contextDataViewSpec.id
+    );
+    if (!existedDataView) {
+      missedIndexPatterns.push({
+        id: contextDataViewSpec.id!,
+        name: contextDataViewSpec.name,
+        title: contextDataViewSpec.title!,
+      });
+    }
+  }
 
   return indexPatterns
     .concat(
-      Object.values(adHocDataViews || {}).map((dataViewSpec) => ({
+      missedIndexPatterns.map((dataViewSpec) => ({
         id: dataViewSpec.id!,
         name: dataViewSpec.name,
         title: dataViewSpec.title!,
```

Comment:
In the `refreshExistingFields` function in this file further down below it's checking for the `spec` property and adds it to the request if it exists. that should only happen for ad-hoc data views (using the new `isPersisted` property): https://github.com/elastic/kibana/blob/089dc75f68e7420b092c60d8a2fc454d79b95740/x-pack/plugins/lens/public/indexpattern_service/loader.ts#L275